pkg/operatorsource/configuring_test.go

```diff
@@ -34,19 +35,20 @@ func TestReconcile_NotConfigured_NewCatalogConfigSourceObjectCreated(t *testing.
 		Message: phase.GetMessage(phase.Succeeded),
 	}
 
+	scheme := runtime.NewScheme()
+	marketplace.AddToScheme(scheme)
+
 	datastore := mocks.NewDatastoreWriter(controller)
-	kubeclient := mocks.NewKubeClient(controller)
+	fakeclient := fake.NewFakeClientWithScheme(scheme)
 
-	reconciler := operatorsource.NewConfiguringReconciler(helperGetContextLogger(), datastore, kubeclient)
+	reconciler := operatorsource.NewConfiguringReconciler(helperGetContextLogger(), datastore, fakeclient)
 
 	ctx := context.TODO()
 	opsrcIn := helperNewOperatorSourceWithPhase("marketplace", "foo", phase.Configuring)
 	namespacedName := types.NamespacedName{Name: "opsrc-foo", Namespace: "marketplace"}
 
 	// We expect that the given CatalogConfigSource object does not exist.
 	cscGet := helperNewCatalogSourceConfig(opsrcIn.Namespace, getExpectedCatalogSourceConfigName(opsrcIn.Name))
-	kubeClientErr := k8s_errors.NewNotFound(schema.GroupResource{}, "CatalogSourceConfig not found")
-	kubeclient.EXPECT().Get(context.TODO(), namespacedName, cscGet).Return(kubeClientErr)
 
 	packages := "a,b,c"
 	datastore.EXPECT().GetPackageIDs().Return(packages)
```

Comment:
The mix of both `fakeClient` and `gomocked` interfaces ( like `datastore` above ) in the same test introduces inconsistency.

Reply:
I think the best approach would be to have all `gomocked` interface calls in one section and all `fakeClient` calls in another. Based on how we're using `gomocked` interfaces and how others have used `fakeClient` in tests, I suggest that all `gomocked` tests should be called before the actual function being tested, and all `fakeClient` calls should be after.